src/ai/backend/agent/agent.py

```diff
@@ -816,7 +817,11 @@ async def _scan_images_wrapper(self, interval: float) -> None:
         self.images = await self.scan_images()
 
     @abstractmethod
-    async def pull_image(self, image_ref: ImageRef, registry_conf: ImageRegistry) -> None:
+    async def pull_image(
+        self, 
+        image_ref: ImageRef, 
+        registry_conf: ImageRegistry, 
+        reporter: Callable[[Union[int, float], Union[int, float]], Awaitable[None]]) -> None:
```

Comment:
For multi-line annotated function arguments, break the last line like this:
```suggestion
        reporter: Callable[[Union[int, float], Union[int, float]], Awaitable[None]],
    ) -> None:
```